File: sharedComposables/src/commonMain/kotlin/dev/mateusz1913/f1results/composable/common/ConstructorStandingRow.kt
package dev.mateusz1913.f1results.composable.common

import androidx.compose.foundation.layout.*
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import dev.mateusz1913.f1results.datasource.data.standings.ConstructorStandingType

@Composable
fun ConstructorStandingRow(constructorStanding: ConstructorStandingType) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Row(
            modifier = Modifier.weight(1f),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            StandingPosition(constructorStanding.position)
            Column(modifier = Modifier.weight(1f)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Text(
                        constructorStanding.constructor.name,
                        fontSize = 18.sp,
                        fontWeight = FontWeight.SemiBold,
                        modifier = Modifier.padding(
                            start = 10.dp,
                            top = 2.dp,
                            end = 10.dp,
                            bottom = 2.dp
                        )
                    )
                }
            }
        }
        StandingScore(constructorStanding.points, constructorStanding.wins)
    }
}

File: swiftApp/Shared/Presentation/Common/ConstructorStandingRow.swift
import SwiftUI
import shared

struct ConstructorStandingRow: View {
    var constructorStanding: ConstructorStandingType
    
    var body: some View {
        HStack(alignment: .center) {
            HStack(alignment: .center) {
                StandingPosition(position: constructorStanding.position)
                VStack {
                    HStack(alignment: .center) {
                        Text(constructorStanding.constructor.name)
                            .font(.system(size: 18))
                            .fontWeight(.semibold)
                            .padding(.init(top: 2, leading: 10, bottom: 2, trailing: 10))
                            .frame(minWidth: 0, maxWidth: .infinity, alignment: .leading)
                    }
                    .fillMaxWidth()
                }
                .fillMaxSize()
            }
            .fillMaxSize()
            StandingScore(points: constructorStanding.points, wins: constructorStanding.wins)
        }
        .fillMaxWidth()
    }
}
